237 student side dashboard media queries

Vfrid opened the pull request:
### Changes
(All changes, unless otherwise specified, were done for both teacher and student sides)
_Everything is under issue #237_
- Changed how big the sidebar is so that on smaller screens it doesn't take up all the space
- Fixed a bug with the sidebar animation breaking on the new dynamic sidebar width
- Fixed size of text on new sidebar so that it always fits but isn't too small
- Fixed header being off-center and components of the header being on top of each other
- Reversed color of light/dark switch b/c it was invisible
- Switched dashboard to a grid layout
- Fixed all parts within dashboard cards (text and div sizes/proportions) for media queries
- Spacing between dashboard cards to be even and card div to be centered (but each new card appears on the left, not in the center of a new row)

Vfrid committed:
starting to edit sizes of teacher dashboard cards

Vfrid committed:
3/20/25

need to fix alignment of cards in teacher dashboard

Vfrid committed:
3/24/25

trying to fix card formatting-- I think I made it better for normal desktop size (on teacher side, didnt update student side yet) but it broke what i did for media queries so ill fix that later

Vfrid committed:
3/25/25

changed flex to grid to fix dashboard card orientation

Vfrid committed:
Update Sidebar.vue

Vfrid committed:
3/28/25

almost finished teacher dashboard media queries (will copy changes over to students when done), just have to fix that students is a tiny bit cut off when viewing from one of the phones

changes to sidebar introduced a bug w transition, now it's choppy. idk what that's about but it's low priority

Vfrid committed:
fdsa

Vfrid committed:
Merge branch 'dev' into 237-student-side-dashboard-media-queries

Vfrid committed:
trying to fix merge

Vfrid committed:
fixed merge

Vfrid committed:
student side media queries

fixed most of it, but have to fix an issue with the assignments stretching vertically if there's more than one

Vfrid committed:
trying to fix assignments in classes on student dashboard

wasted a lot of time fixing an issue running the app but was able to start fixing the assignments to fit properly, need to fix issue with overflow still

Vfrid committed:
fixed overflow issue, fixed sidebar link size

Vfrid committed:
finished teacher and student dashboard displays

Vfrid committed:
small student side text-size edit

Vfrid committed:
trying to fix animation

i broke it a long time ago when i fixed the sidebar and i just realized

Vfrid committed:
fixed sidebar animation!!

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -1,33 +1,33 @@
   <NuxtLink
     :to="`/student/course/${course.id}`"
-    class="flex w-115 flex-col items-center justify-center overflow-hidden rounded-xl border border-neutral-300 bg-body transition hover:border-neutral-600/50 hover:shadow-lg dark:border-neutral-600 dark:hover:border-neutral-300/50"
+    class="flex my-2 flex h-[calc(100%)] w-[calc(85%)] flex-col items-center justify-center overflow-hidden rounded-xl border border-neutral-300 bg-body transition hover:border-neutral-600/50 hover:shadow-lg dark:border-neutral-600 dark:hover:border-neutral-300/50"
```
h-full
you dont need calc for everything

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -1,33 +1,33 @@
   >
     <!-- course information -->
-    <div class="flex h-24 w-full flex-col items-center justify-end p-2" :style="{ backgroundColor: subjectColors[course.subject] }">
+    <div class="flex lg:h-24 md:h-24 h-2/5 w-full flex-col items-center justify-end p-2" :style="{ backgroundColor: subjectColors[course.subject] }">
```
useless media query

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -1,33 +1,33 @@
     </div>
 
-    <div class="flex h-full min-h-36 w-full flex-col items-center justify-start p-2">
+    <div class="flex h-full min-h-38 w-full flex-col items-center justify-start p-2">
```
doesnt exist

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -1,33 +1,33 @@
           :key="assignment.id"
           :to="`/student/course/${course.id}/${assignment.id}`"
-          class="flex h-full w-full flex-col items-center justify-center rounded-xl border border-neutral-300 p-3 hover:shadow-lg dark:border-neutral-600 dark:hover:border-neutral-300/50"
+          class="flex lg:h-24 md:h-30 h-34 overflow-hidden w-full flex-col items-center justify-around rounded-xl border border-neutral-300 p-3 hover:shadow-lg dark:border-neutral-600 dark:hover:border-neutral-300/50"
```
doesnt exist
doesnt exist

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -1,33 +1,33 @@
 
           <div class="flex h-full w-full flex-col items-center justify-start gap-3">
-            <p class="w-64 overflow-hidden overflow-ellipsis text-nowrap text-center text-xl font-semibold">{{ assignment.assignment.name }}</p>
+            <p class="w-64 overflow-hidden overflow-ellipsis text-nowrap text-center text-[calc(85%)] lg:text-[calc(120%)] md:text-[calc(98%)] font-semibold -mb-3">{{ assignment.assignment.name }}</p>
```
why not use lg xl 2xl etc?
whats the point of 98%?? is the 2% really that big of a difference??

kennething commented on components/student/Dashboard/Header.vue:
```diff
@@ -12,10 +12,10 @@
-        <img class="size-12 rounded-full transition duration-300 group-active:scale-95" src="/siths.png" aria-hidden="true" />
-        <span class="text-2xl font-semibold transition duration-300 group-hover:translate-x-0.5">Prep</span>
+      <NuxtLink to="/student/dashboard" class="group flex items-center lg:justify-center md:justify-center justify-end gap-2">
+        <span class="text-l lg:text-2xl md:text-xl font-semibold transition duration-300 group-hover:-translate-x-0.5">Regents</span>
```
doesnt exist

kennething commented on components/student/Dashboard/Header.vue:
```diff
@@ -12,10 +12,10 @@
+      <NuxtLink to="/student/dashboard" class="group flex items-center lg:justify-center md:justify-center justify-end gap-2">
+        <span class="text-l lg:text-2xl md:text-xl font-semibold transition duration-300 group-hover:-translate-x-0.5">Regents</span>
+        <img class="size-12 rounded-full transition duration-300 group-hover:shadow-md group-active:scale-95" src="/siths.png" aria-hidden="true" />
+        <span class="text-l lg:text-2xl md:text-xl font-semibold transition duration-300 group-hover:translate-x-0.5">Prep</span>
```
doesnt exist

Vfrid replied:
do you mean you just got rid of the entire div on dev? or is it something about my changes??

kennething replied:
text-l is not a thing

Vfrid replied:
oh damn ok sorry that was a typo im just blind ig

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,25 @@
-      <p>{{ course.teacher }}</p>
-      <p v-if="!course.teacher">{{ userStore.name }}</p>
+    <div class="flex h-1/2 w-full flex-col items-center justify-end pt-2 px-2 " :style="{ backgroundColor: subjectColors[course.subject] }">
+      <h2 :title="course.name" class="sm:text-l mt-4 mb-1 h-1/2 w-full overflow-hidden overflow-ellipsis text-nowrap text-center font-semibold md:text-2xl lg:text-3xl self-center -mt-3 -mb-2">{{ course.name }}</h2>
```
doesnt exist
conflicting margin classes

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,25 @@
+    <div class="flex h-1/2 w-full flex-col items-center justify-end pt-2 px-2 " :style="{ backgroundColor: subjectColors[course.subject] }">
+      <h2 :title="course.name" class="sm:text-l mt-4 mb-1 h-1/2 w-full overflow-hidden overflow-ellipsis text-nowrap text-center font-semibold md:text-2xl lg:text-3xl self-center -mt-3 -mb-2">{{ course.name }}</h2>
+      <p class="text-sm lg:text-lg md:text-lg">Period {{ course.period }}</p>
+      <p class="text-center lg:text-lg md:text-lg text-sm">{{ course.teacher }}</p>
```
useless media query

Vfrid replied:
what specifically are you saying is useless? it works. I tested with and without and it's better with...

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,25 @@
-      <p v-if="!course.teacher">{{ userStore.name }}</p>
+    <div class="flex h-1/2 w-full flex-col items-center justify-end pt-2 px-2 " :style="{ backgroundColor: subjectColors[course.subject] }">
+      <h2 :title="course.name" class="sm:text-l mt-4 mb-1 h-1/2 w-full overflow-hidden overflow-ellipsis text-nowrap text-center font-semibold md:text-2xl lg:text-3xl self-center -mt-3 -mb-2">{{ course.name }}</h2>
+      <p class="text-sm lg:text-lg md:text-lg">Period {{ course.period }}</p>
```
useless media query

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,25 @@
+      <h2 :title="course.name" class="sm:text-l mt-4 mb-1 h-1/2 w-full overflow-hidden overflow-ellipsis text-nowrap text-center font-semibold md:text-2xl lg:text-3xl self-center -mt-3 -mb-2">{{ course.name }}</h2>
+      <p class="text-sm lg:text-lg md:text-lg">Period {{ course.period }}</p>
+      <p class="text-center lg:text-lg md:text-lg text-sm">{{ course.teacher }}</p>
+      <p v-if="!course.teacher" class="text-center lg:text-lg md:text-lg text-sm mb-1">{{ userStore.name }}</p>
```
useless media query

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,25 @@
-      <div class="w-full">
-        <div class="h-1/2 text-center text-2xl font-semibold"><h3 class="self-center pt-3">Join Code</h3></div>
-        <div class="h-1/2 text-center text-2xl">
+    <div class="flex h-full min-h-24 w-full columns-2 rounded-3xl bg-[var(--bg-color)] p-2 ">
```
use bg-body

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,25 @@
-        <div class="h-1/2 text-center text-2xl font-semibold"><h3 class="self-center pt-3">Join Code</h3></div>
-        <div class="h-1/2 text-center text-2xl">
+    <div class="flex h-full min-h-24 w-full columns-2 rounded-3xl bg-[var(--bg-color)] p-2 ">
+      <div class="lg:w-full md:w-full w-[calc(85%)]">
```
useless media query

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,25 @@
-        <div class="h-1/2 text-center text-2xl">
+    <div class="flex h-full min-h-24 w-full columns-2 rounded-3xl bg-[var(--bg-color)] p-2 ">
+      <div class="lg:w-full md:w-full w-[calc(85%)]">
+        <div class="sm:text-l h-1/2 text-center font-semibold md:text-xl lg:text-2xl sm:mb-1"><h3 class="self-center pt-5">Join Code</h3></div>
```
doesnt exist

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,25 @@
       <div class="h-20 w-[1px] bg-neutral-300 dark:bg-neutral-600"></div>
       <div class="w-full">
-        <div class="h-full content-center text-center text-2xl">
+        <div class="h-full content-center text-center sm:text-xl md:text-2xl lg:text-2xl">
```
useless media query
md and lg are applying the same thing

kennething commented on components/teacher/Dashboard/Header.vue:
```diff
@@ -12,14 +12,14 @@
     <div class="flex items-center justify-center">
-      <NuxtLink to="/teacher/dashboard" class="group flex items-center justify-center gap-2">
-        <span class="text-2xl font-semibold transition duration-300 group-hover:-translate-x-0.5">Regents</span>
+      <NuxtLink to="/teacher/dashboard" class="group flex items-center lg:justify-center md:justify-center justify-end gap-2">
```
useless media query

kennething commented on components/teacher/Dashboard/Header.vue:
```diff
@@ -12,14 +12,14 @@
-      <NuxtLink to="/teacher/dashboard" class="group flex items-center justify-center gap-2">
-        <span class="text-2xl font-semibold transition duration-300 group-hover:-translate-x-0.5">Regents</span>
+      <NuxtLink to="/teacher/dashboard" class="group flex items-center lg:justify-center md:justify-center justify-end gap-2">
+        <span class="text-l lg:text-2xl md:text-xl font-semibold transition duration-300 group-hover:-translate-x-0.5">Regents</span>
```
doesnt exist

kennething commented on components/teacher/Dashboard/Header.vue:
```diff
@@ -12,14 +12,14 @@
+        <span class="text-l lg:text-2xl md:text-xl font-semibold transition duration-300 group-hover:-translate-x-0.5">Regents</span>
         <img class="size-12 rounded-full transition duration-300 group-hover:shadow-md group-active:scale-95" src="/siths.png" aria-hidden="true" />
-        <span class="text-2xl font-semibold transition duration-300 group-hover:translate-x-0.5">Prep</span>
+        <span class="text-l lg:text-2xl md:text-xl font-semibold transition duration-300 group-hover:translate-x-0.5">Prep</span>
```
doesnt exist

kennething commented on pages/student/dashboard.vue:
```diff
@@ -1,9 +1,9 @@
 <template>
-  <div class="flex w-full flex-col items-start justify-start">
-    <div v-if="loaded" class="w-full">
+  <div class="flex h-full w-full flex-col items-center justify-center mx-auto">
```
mx-auto does nothing

kennething commented on pages/student/dashboard.vue:
```diff
@@ -1,9 +1,9 @@
-      <div class="flex w-full flex-col">
-        <div v-if="studentCourses.length > 0" class="flex flex-wrap content-start items-start justify-around gap-8">
+      <div class="flex h-full w-full flex-col justify-around overflow-hidden">
+        <div v-if="studentCourses.length > 0" class="mx-auto grid w-full grid-cols-1 justify-center items-center md:grid-cols-2 lg:grid-cols-2 place-items-center gap-y-5">
```
useless media query

kennething commented on pages/teacher/dashboard.vue:
```diff
@@ -1,16 +1,18 @@
 <template>
-  <div class="flex h-full w-full flex-col items-start justify-start">
+  <div class="flex h-full w-full flex-col items-center justify-center mx-auto">
```
mx-auto does nothing here

kennething reviewed (changes requested):
- i would recommend not using notepad for code editing. please use extensions including the extensions specifically designed for vue and tailwind (this is a demand not a request) (also use prettier)
- you dont need [calc(%)] to do %s 
- try to use standard classes (lg, xl, 2xl) instead of arbitrary values whenever possible
- its impressive how chatgpt would write better code than this
- ~~Reversed color of light/dark switch b/c it was invisible~~ :x:

Vfrid committed:
fixing requested pr changes

Vfrid committed:
fixing pr changes

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -1,7 +1,7 @@
   <NuxtLink
     :to="`/student/course/${course.id}`"
-    class="flex w-115 flex-col items-center justify-center overflow-hidden rounded-xl border border-neutral-300 bg-body transition hover:border-neutral-600/50 hover:shadow-lg dark:border-neutral-600 dark:hover:border-neutral-300/50"
+    class="my-2 flex h-full w-[calc(85%)] flex-col items-center justify-center overflow-hidden rounded-xl border border-neutral-300 bg-body transition hover:border-neutral-600/50 hover:shadow-lg dark:border-neutral-600 dark:hover:border-neutral-300/50"
```
unnecessary calc
just do w-[85%]

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -11,23 +11,23 @@
     </div>
 
-    <div class="flex h-full min-h-36 w-full flex-col items-center justify-start p-2">
+    <div class="min-h-38 flex h-full w-full flex-col items-center justify-start p-2">
```
min-h-38 still does not exist

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -11,23 +11,23 @@
       <h3 class="pb-2 pt-1 text-xl font-bold">Assignments</h3>
 
-      <div v-if="assignments.length > 0" class="flex h-full w-full flex-wrap items-start justify-around gap-3 px-3 pb-3">
+      <div v-if="assignments.length > 0" class="flex h-full w-full flex-wrap items-start justify-around gap-3 overflow-hidden px-3 lg:overflow-hidden">
```
still useless media query

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -11,23 +11,23 @@
           :key="assignment.id"
           :to="`/student/course/${course.id}/${assignment.id}`"
-          class="flex h-full w-full flex-col items-center justify-center rounded-xl border border-neutral-300 p-3 hover:shadow-lg dark:border-neutral-600 dark:hover:border-neutral-300/50"
+          class="md:h-30 h-34 flex w-full flex-col items-center justify-around overflow-hidden rounded-xl border border-neutral-300 p-3 hover:shadow-lg lg:h-24 dark:border-neutral-600 dark:hover:border-neutral-300/50"
```
h-30 and h-34 still do not exist

kennething commented on components/student/Dashboard/DashboardCard.vue:
```diff
@@ -11,23 +11,23 @@
 
           <div class="flex h-full w-full flex-col items-center justify-start gap-3">
-            <p class="w-64 overflow-hidden overflow-ellipsis text-nowrap text-center text-xl font-semibold">{{ assignment.assignment.name }}</p>
+            <p class="-mb-3 w-64 overflow-hidden overflow-ellipsis text-nowrap text-center text-lg font-semibold md:text-xl lg:text-xl">{{ assignment.assignment.name }}</p>
```
still useless media query
md and lg are applying the same thing

kennething commented on components/student/Dashboard/Header.vue:
```diff
@@ -12,10 +12,10 @@
-        <span class="text-2xl font-semibold transition duration-300 group-hover:-translate-x-0.5">Regents</span>
-        <img class="size-12 rounded-full transition duration-300 group-active:scale-95" src="/siths.png" aria-hidden="true" />
-        <span class="text-2xl font-semibold transition duration-300 group-hover:translate-x-0.5">Prep</span>
+      <NuxtLink to="/student/dashboard" class="group flex items-center justify-end gap-2 md:justify-center lg:justify-center">
```
still useless media query
md and lg are applying the same thing

kennething commented on components/student/Dashboard/Sidebar/PageLink.vue:
```diff
@@ -1,7 +1,7 @@
   <NuxtLink
     :to="to"
-    class="flex h-12 w-full items-center justify-start gap-3 rounded-xl pl-4 text-xl hover:bg-neutral-200/50 dark:hover:bg-neutral-500/25"
+    class="flex h-12 w-full items-center justify-start gap-3 rounded-xl pl-4 text-[calc(100%)] hover:bg-neutral-200/50 dark:hover:bg-neutral-500/25"
```
unnecessary calc
and use a default tailwind class for text size instead of 100%

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,27 @@
 <template>
   <div
-    class="flex w-115 flex-col items-center justify-center overflow-hidden rounded-xl border border-neutral-300 bg-body transition hover:border-neutral-600/50 hover:shadow-lg dark:border-neutral-600 dark:hover:border-neutral-300/50"
+    class="my-2 flex h-[calc(90%)] w-[calc(85%)] flex-col items-center justify-center overflow-hidden rounded-xl border border-neutral-300 bg-body transition hover:border-neutral-600/50 hover:shadow-lg dark:border-neutral-600 dark:hover:border-neutral-300/50"
```
unnecessary calcs

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,27 @@
+      <h2 :title="course.name" class="my-1 h-1/2 w-full self-center overflow-hidden overflow-ellipsis text-nowrap text-center text-lg font-semibold md:text-2xl lg:text-3xl">
+        {{ course.name }}
+      </h2>
+      <p class="text-sm md:text-lg lg:text-lg">Period {{ course.period }}</p>
```
still useless media query
md and lg are applying the same thing

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,27 @@
+        {{ course.name }}
+      </h2>
+      <p class="text-sm md:text-lg lg:text-lg">Period {{ course.period }}</p>
+      <p class="text-center text-sm md:text-lg lg:text-lg">{{ course.teacher }}</p>
```
still useless media query
md and lg are applying the same thing

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,27 @@
+      </h2>
+      <p class="text-sm md:text-lg lg:text-lg">Period {{ course.period }}</p>
+      <p class="text-center text-sm md:text-lg lg:text-lg">{{ course.teacher }}</p>
+      <p v-if="!course.teacher" class="mb-1 text-center text-sm md:text-lg lg:text-lg">{{ userStore.name }}</p>
```
vstill useless media query
md and lg are applying the same thing

kennething commented on components/teacher/Dashboard/DashboardCard.vue:
```diff
@@ -1,26 +1,27 @@
-        <div class="h-1/2 text-center text-2xl font-semibold"><h3 class="self-center pt-3">Join Code</h3></div>
-        <div class="h-1/2 text-center text-2xl">
+    <div class="flex h-full min-h-24 w-full columns-2 rounded-3xl bg-body p-2">
+      <div class="w-[calc(85%)] md:w-full lg:w-full">
```
unnecessary calc
still useless media query
md and lg are applying the same thing

kennething commented on components/teacher/Dashboard/Header.vue:
```diff
@@ -12,14 +12,14 @@
     <div class="flex items-center justify-center">
-      <NuxtLink to="/teacher/dashboard" class="group flex items-center justify-center gap-2">
-        <span class="text-2xl font-semibold transition duration-300 group-hover:-translate-x-0.5">Regents</span>
+      <NuxtLink to="/teacher/dashboard" class="group flex items-center justify-end gap-2 md:justify-center lg:justify-center">
```
still useless media query
md and lg are applying the same thing

kennething commented on components/teacher/Dashboard/Sidebar/PageLink.vue:
```diff
@@ -1,7 +1,7 @@
   <NuxtLink
     :to="to"
-    class="flex h-12 w-full items-center justify-start gap-3 rounded-xl pl-4 text-xl hover:bg-neutral-200/50 dark:hover:bg-neutral-500/25"
+    class="flex h-12 w-full items-center justify-start gap-3 rounded-xl pl-4 text-[calc(100%)] hover:bg-neutral-200/50 dark:hover:bg-neutral-500/25"
```
unnecessary calc
and use a default tailwind class for text size instead of 100%

kennething commented on pages/student/dashboard.vue:
```diff
@@ -1,9 +1,9 @@
-      <div class="flex w-full flex-col">
-        <div v-if="studentCourses.length > 0" class="flex flex-wrap content-start items-start justify-around gap-8">
+      <div class="flex h-full w-full flex-col justify-around overflow-hidden">
+        <div v-if="studentCourses.length > 0" class="mx-auto grid w-full grid-cols-1 place-items-center items-center justify-center gap-y-5 md:grid-cols-2 lg:grid-cols-2">
```
still useless media query
md and lg are applying the same thing

kennething commented on package.json:
```diff
@@ -10,11 +10,13 @@
   "dependencies": {
     "@pinia/nuxt": "^0.7.0",
     "flyonui": "^1.3.0",
+    "lint": "^1.1.2",
```
??

Vfrid replied:
when I pushed my commit yesterday it said it failed the run for prettier and lint. i was pretty sure i had prettier already but i guess i didnt, and i'd never heard of lint before. since you told me to get prettier and said extension **s** in your original message i assumed you wanted all things i had failed, so i downloaded lint. sorry, ill take it off. my bad

for the rest of the changes i just didnt get to it yet, my computer sucks so i cant work on it at home, im just making my way through it during school.

kennething replied:
[prettier vscode extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)

Vfrid replied:
i've had it since before I ever sent in the PR...

kennething reviewed (changes requested):
dont install random packages ??

Vfrid committed:
done fixing changes????

idk what the deal w prettier is bc ive had the vs code extension always... but i got rid of lint (kept eslint though, i need that right??????? idek anymore)

kennething commented on components/teacher/Dashboard/Sidebar/PageLink.vue:
```diff
@@ -1,7 +1,7 @@
   <NuxtLink
     :to="to"
-    class="flex h-12 w-full items-center justify-start gap-3 rounded-xl pl-4 text-xl hover:bg-neutral-200/50 dark:hover:bg-neutral-500/25"
+    class="text-md flex h-12 w-full items-center justify-start gap-3 rounded-xl pl-4 hover:bg-neutral-200/50 md:text-lg lg:text-xl dark:hover:bg-neutral-500/25"
```
text-md doesnt exist

kennething committed:
unmerge conflict

kennething committed:
ts pmo icl

kennething committed:
Merge branch 'dev' into 237-student-side-dashboard-media-queries

Vfrid committed:
done

bridgetf7 reviewed (changes requested):
please have the sidebar be closed by default on mobile and small screens

Vfrid committed:
sidebar auto-closed on small screens

bridgetf7 reviewed (changes requested):
![image](https://github.com/user-attachments/assets/3eb6d483-bbb3-4227-bd97-12171e717855)
Side bar looks awkward on phone screen (I would have it take up 50-80% of the screen)
![image](https://github.com/user-attachments/assets/0211318c-3280-4aa4-afbb-c4d3cd2d5a09)
There's an overlap of components here; also snap the entire line of components to the left hand side 
![image](https://github.com/user-attachments/assets/f20019a1-16cf-4bde-8a44-dc2ad6ef8e1a)
The sidebar doesn't automatically close when switching links 
![image](https://github.com/user-attachments/assets/53bc29b0-190a-4265-823a-0fe33647cb4e)
I would make this font slightlyyyy bigger; around `text-2xl` would be around right 
![image](https://github.com/user-attachments/assets/9daa8627-a917-47f5-8e6e-07f23dd8efd8)

Vfrid committed:
fixed size of sidebar student-side

still need to fix progress bar thing and the other issues bridget requested

Vfrid committed:
progress bar fixed, search fixed

have to fix the other changes still, i started working on closing side bar automatically when u switch pages

Vfrid committed:
umm what

sorry i didnt solve the todo sidebar thing today it's being really weird

Vfrid committed:
fixed sidebar, working on making the todo page fit nicer

for some reason the sidebar height was 100dvh MINUS 4rem, and it was making it not descend all the way to the bottom of the page. I got rid of that but if there's some reason I'm not understanding for it pls lmk

Vfrid committed:
done with everything except teacher-side class page

I chose not to snap the todo toolbar to the left b/c I tried it and I think it looks worse (I got a second opinion too). If you still want that, lmk and I will change it.

Vfrid committed:
done?
